core/trino-main/src/main/java/io/trino/operator/TopNOperator.java

```diff
@@ -209,17 +229,20 @@ private void addPage(Page page)
                 return TransformationState.needsMoreData();
             }
```

Comment:
We should not return needsMoreData() if adding the page moves us to a flushing state?

Reply:
It seems that it will never be true. So, let me utilize this for partial flushing.

Reply:
I had a deep dive into the implementation and it seems that this code is a dead code. Let me utilize it for partial flushing.

Reply:
Done